### PR TITLE
add support for node labels for managed clusters

### PR DESCRIPTION
**Pull request by zmalik**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes part of #1701

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
add support for node labels in AKS agent pool
```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/ok-to-test

File: azure/scope/managedcontrolplane.go [hunk 1]
```diff
@@ -558,6 +558,7 @@ func (s *ManagedControlPlaneScope) AgentPoolSpec() azure.AgentPoolSpec {
 		),
 		Mode:              s.InfraMachinePool.Spec.Mode,
 		AvailabilityZones: s.InfraMachinePool.Spec.AvailabilityZones,
+		NodeLabels:        s.InfraMachinePool.Spec.NodeLabels,
```
Comment (CecileRobertMichon) [inline azure/scope/managedcontrolplane.go:561]: does this need to be added above in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1927/files#diff-cfd94adbc2ff767742e457e64836e93e88c6e03c3a64c04619c54b7ed76f284eR503 ?

Comment (zmalik) [reply azure/scope/managedcontrolplane.go:561]: yes! I will do that for rest of PRs also

File: exp/api/v1beta1/azuremanagedmachinepool_types.go [hunk 2]
```diff
@@ -58,6 +58,9 @@ type AzureManagedMachinePoolSpec struct {
 	// +optional
 	AvailabilityZones []string `json:"availabilityZones,omitempty"`
 
+	// Node labels - labels for all of the nodes present in node pool
+	NodeLabels map[string]*string `json:"nodeLabels,omitempty"`
```
Comment (CecileRobertMichon) [inline exp/api/v1beta1/azuremanagedmachinepool_types.go:62]: should this be marked as optional?

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

File: azure/types.go [hunk 3]
```diff
@@ -409,6 +409,9 @@ type AgentPoolSpec struct {
 	// Minimum number of nodes for auto-scaling
 	MinCount *int32 `json:"minCount,omitempty"`
 
+	// Node labels - labels for all of the nodes present in node pool
+	NodeLabels map[string]*string `json:"nodeLabels,omitempty"`
```
Comment (jackfrancis) [inline azure/types.go:413]: What's the value of the `*string` value here? Do we really want to permit a key string with a nil value?

Comment (zmalik) [reply azure/types.go:413]: I will change it. The initial motivation was to avoid conversion from `map[string]string` to `map[string]*string`

Comment (zmalik) [reply azure/types.go:413]: done

**Comment by CecileRobertMichon** [comment]
@zmalik can you please squash commits?

@jackfrancis PTAL

**Comment by CecileRobertMichon** [comment]
lgtm

**Comment by zmalik** [comment]
rebased and squashed

File: azure/types.go (hunk 3 above)
Comment (zmalik) [reply azure/types.go:367]: wait, not done but I changed it in `azuremanagedmachinepool_types.go` as in our API types yes I agree that there is no reason to use `*string`
but ehere I kept the same as the one expected in `ManagedClusterAgentPoolProfileProperties`

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by CecileRobertMichon** [comment]
>  getting a new SSH client connection: ssh: handshake failed: ssh: unable to authenticate

/retest

**Pull request merged**